### ISSUE TITLE
Feature to collect profiling data on main thread only

### ISSUE THREAD

**Issue by boxed** [benfred/py-spy#175]
I get a lot of irrelevant stuff due to sentry_sdk having a thread that is basically idling and other similar stuff. Intuitively it seems like it could also be faster to just collect for the main thread which is a nice side effect if that's what you want.

**Comment by benfred** [comment]
We have an option already to only profile threads holding the GIL (```--gil```).  Have you tried this out?

Also what OS/version of py-spy are you using? We should be doing a much better job of filtering out idle frames with py-spy v0.2.0 (for x86_64 linux/windows/osx anyways).

**Comment by boxed** [comment]
Well I want to see the time ww spend waiting for the db so I already use --idle. So I guess that means using --gil would be counterproductive right?

Linux/0.2.something. Dont remember the exact day I downloaded :P

**Comment by benfred** [comment]
Yeah - using --gil with --idle is pretty counterproductive (and probably should throw an error if you try to do that).

In the meantime, if you're generating a flamegraph you can specify the ```--threads``` option to root each trace with it's thread id. You can either zoom into the main thread there, or record to a 'raw' format and then filter out the other threads before passing to the flamegraph.pl script.  I realize that isn't a perfect solution though =(

**Comment by boxed** [comment]
Ah. That's good to know though!

**Comment by benfred** [comment]
@codypiersall  I'm not sure what the best default here is, I'm tempted to leave as is for now just to avoid disruption.  

I'd like to make it so that people can customize py-spy to default as they'd like, but thats a way's off right now. In the meantime if you define a function like this in your .bashrc then this should add the --threads flag to each ```py-spy record``` invocation:

```
py-spy() {
    if [[ $1 == "record" ]]; then
        shift
        command py-spy record --threads $@
    else
        command py-spy $@
    fi
}
```
```

**Comment by olejorgenb** [comment]
Related: https://github.com/benfred/py-spy/issues/746
